src/modm/processing/fiber/scheduler.hpp.in

```diff
@@ -100,6 +118,35 @@ class Scheduler
 	}
 };
 
+%% if multicore
+%% for i in range(num_cores)
+template <>
+{{core_storage[i]}} Fiber<{{i}}>* Scheduler<{{i}}>::last = nullptr;
+template <>
+{{core_storage[i]}} Fiber<{{i}}>* Scheduler<{{i}}>::current = nullptr;
+%% endfor
+template <size_t Core>
+inline void
+yieldCore()
+{
+	if (Scheduler<Core>::current == nullptr) return;
+	Fiber<Core>* next = Scheduler<Core>::current->next;
+	if (next == Scheduler<Core>::current) return;
+	Scheduler<Core>::last = Scheduler<Core>::current;
+	Scheduler<Core>::current->jump(*next);
+}
+inline void
+yield()
+{
+	using yield_func = void(*)();
+	static constexpr yield_func yields[] = {
+%% for i in range(num_cores)
+		&yieldCore<{{i}}>,
+%% endfor
+	};
+	yields[multicore::Core::getNum()]();
```

Comment:
Yes, exactly, but at a Scheduler level (requires some refactoring for a static getter function, sorry).